src/components/sites-sync-modal-selector.tsx

```diff
@@ -0,0 +1,213 @@
+import { SearchControl } from '@wordpress/components';
+import { sprintf } from '@wordpress/i18n';
+import { useI18n } from '@wordpress/react-i18n';
+import { useState } from 'react';
+import { SyncSite } from '../hooks/use-sync-sites';
+import { cx } from '../lib/cx';
+import { getIpcApi } from '../lib/get-ipc-api';
+import { Badge } from './badge';
+import Button from './button';
+import Modal from './modal';
+import { WordPressShortLogo } from './wordpress-short-logo';
+
+export function SitesSyncModalSelector( {
+	onRequestClose,
+	onConnect,
+	syncSites,
+}: {
+	onRequestClose: () => void;
+	syncSites: SyncSite[];
+	onConnect: ( siteId: number ) => void;
+} ) {
+	const { __ } = useI18n();
+	const [ selectedSiteId, setSelectedSiteId ] = useState< number | null >( null );
+	const [ searchQuery, setSearchQuery ] = useState< string >( '' );
+	const filteredSites = syncSites.filter( ( site ) =>
+		site.name.toLowerCase().includes( searchQuery.toLowerCase() )
+	);
+	return (
+		<Modal
+			className="w-3/5 h-full max-h-[84vh] [&>div]:!p-0"
+			onRequestClose={ onRequestClose }
+			title={ __( 'Connect a WordPress.com site' ) }
+		>
+			<SearchSites searchQuery={ searchQuery } setSearchQuery={ setSearchQuery } />
+			<div className="h-[calc(84vh-230px)]">
+				{ filteredSites.length === 0 ? (
+					<div className="flex justify-center items-center h-full">
+						{ searchQuery
+							? sprintf( __( 'No sites found for "%s"' ), searchQuery )
+							: __( 'No sites found' ) }
+					</div>
+				) : (
+					<ListSites
+						syncSites={ filteredSites }
+						selectedSiteId={ selectedSiteId }
+						onSelectSite={ setSelectedSiteId }
+					/>
+				) }
+			</div>
+			<Footer
+				onRequestClose={ onRequestClose }
+				onConnect={ () => {
+					if ( ! selectedSiteId ) {
+						return;
+					}
+					onConnect( selectedSiteId );
+					onRequestClose();
+				} }
+				disabled={ ! selectedSiteId }
+			/>
+		</Modal>
+	);
+}
+
+function SearchSites( {
+	searchQuery,
+	setSearchQuery,
+}: {
+	searchQuery: string;
+	setSearchQuery: ( value: string ) => void;
+} ) {
+	const { __ } = useI18n();
+	return (
+		<div className="flex flex-col px-8 pb-6 border-b border-a8c-gray-5">
+			<SearchControl
+				className="w-full"
+				placeholder={ __( 'Search sites' ) }
```

Comment:
Should we adjust the color to a8c-gray-900 to be consistent with Figma?

<img width="521" alt="Screenshot 2024-10-17 at 4 18 38 PM" src="https://github.com/user-attachments/assets/ff4270a6-fae9-4f19-beb2-219bb67ac317"> 

The top color is from Figma and the bottom one is the current implementation.

Reply:
We can use the lighter grey for the placeholder text. The Figma component we have the SearchControl appears to use the filled text styles for the placeholder text.

Reply:
Sounds good. Let's use the default style component where is possible 💪

Reply:
@katinthehatsite, is it good to merge?

Reply:
@sejas I will take another look in a bit but looked good so far 👍